v1.1/architecture/replication-layer.md

```diff
@@ -53,17 +62,25 @@ After loading the snapshot, the node gets up to date by replaying all actions fr
 
 ### Leases
 
-A single node in the Raft group acts as the Leaseholder, which is the only node that can serve reads or propose writes to the Raft group leader (both actions are received as `BatchRequests` from [`DistSender`](distribution-layer.html#distsender)).
+Enforcing our [isolation levels](transaction-layer.html#isolation-levels) relies on a [timestamp cache](transaction-layer.html#timestamp-cache) that tracks the timestamp of each key's most recent read. If a node receives a write request with a lower timestamp (i.e. "in the past"), allowing it would violate serializable isolation (but not snapshot isolation).
+
+To keep this timestamp cache up to date and consistent, a single node in the Raft group is elected as the Leaseholder. This node serves all reads for the range (ensuring its timestamp cache is always accurate) and proposes all writes to the Raft group leader (except those which it rejects because of failed timestamp cache checks).
+
+{{site.data.alerts.callout_info}}CockroachDB uses a single node's timestamp cache for efficiency's sake. This mechanism could be alternatively designed to use a consensus-based timestamp cache and serve reads from any replica, but each read would incur more network round trips.{{site.data.alerts.end}}
+
+#### Acquiring and Maintaining a Lease
+
+Any replica that can communicate with a majority of the range's replicas can become the Leaseholder. A replica establishes itself as the Leaseholder by committing a special lease acquisition log entry through Raft (which is a write and therefore requires consensus). The Leaseholder maintains the lease by continually updating the lease's expiration (which is another write and requires consensus). If the node were to be partitioned away from a majority of the range's replicas, it would no longer be able to extend its lease's expiration.
```

Comment:
We only update the lease's expiration for a few special ranges (the meta and liveness ranges). For other ranges, the lease includes a reference to the node's liveness record, and we periodically update the expiration of the liveness record (one per node) instead of the leases (one per range). The lease is then considered to be expired when the corresponding node liveness record expires. Atomic test-and-set operations are used to ensure that leases and liveness records stay consistent.